requirements-dev.txt

```diff
@@ -17,4 +17,5 @@ click==8.0.4
 cmake
 custatevec-cu12
 pylint
-scipy
+scipy==1.12.0
```

Comment:
Is scipy pinned because of compatibility with `quimb`?

Reply:
Yes. At some point, there was [an error](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8637854274/job/23681014172) coming from `quimb`. Eventually, I realized it was due to the wrong version of `scipy` (also used by `quimb`).

Reply:
Why is `quimb` pinned?

Reply:
There's no need to pin `quimb` indeed, do I just removed the pin

Reply:
The pin is still there though.